src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -322,6 +326,10 @@ public boolean authenticate(final RestRequest request, final RestChannel channel
                 log.debug("securitytenant '{}'", tenant);
             }
 
+            if (InternalAuthenticationBackend.INTERNAL.equals(authDomain.getBackend().getType())) {
```

Comment:
Shouldn't the internal authenticating backend be responsible for adding custom logic in its implementation?

Reply:
Moved the call to `setInternal` on the user to within the `InternalAuthenticationBackend.authenticate` before it returns the user.